### ISSUE TITLE
Using gray shade from string ignores alpha parameters

### ISSUE THREAD

**Issue by zblz** [matplotlib/matplotlib#6513]
Using matplotlib master and a color defined by a string indicating a gray shade, the alpha parameter gets ignored in calls to, e.g. `plt.plot`. Example:

```
intercepts=np.random.randn(100)

ms=np.random.randn(100)
x=np.linspace(-5,5,100)
for m,i in zip(ms,intercepts):
    plt.plot(m*x+i, c='0.1', alpha=0.2)
```
results in:
![string](https://cloud.githubusercontent.com/assets/576258/15681263/6c9551e8-2750-11e6-82e3-ef5fdc549b2d.png)

whereas the expected result can be obtained by converting the string to a tuple:
```
for m,i in zip(ms,intercepts):
    plt.plot(m*x+i, c=(0.1, 0.1, 0.1, 1.0), alpha=0.2)
```
![tuple](https://cloud.githubusercontent.com/assets/576258/15681302/8aba061e-2750-11e6-840b-a8a0b935fabd.png)

Looking at the code, it seems that all colors are passed through `colors.to_rgba`, and in my setup it correctly converts the string to the tuple, so it seems the problem is elsewhere.
```
>>> mpl.colors.to_rgba('0.1')
(0.1, 0.1, 0.1, 1.0)
```

This used to work in previous v2.x branch commits (cannot really pinpoint where it failed first time) and in v1.5.

**Comment by tacaswell** [comment]
attn @anntzer

**Comment by tacaswell** [comment]
closed by #6517
